File: woodroid/src/main/java/me/gilo/woodroid/models/Customer.java
package me.gilo.woodroid.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;


public class Customer implements Serializable{
    public int id;
    @SerializedName("created_at")
    public String createdAt;
    public String email;
    @SerializedName("first_name")
    public String firstName;
    @SerializedName("last_name")
    public String lastName;
    public String username;
    public String password;
    public String role;
    @SerializedName("last_order_id")
    public String lastOrderId;
    @SerializedName("last_order_date")
    public String lastOrderDate;
    @SerializedName("orders_count")
    public int ordersCount;
    @SerializedName("total_spent")
    public String totalSpent;
    @SerializedName("avatar_url")
    public String avatarUrl;
    @SerializedName("billing")
    public BillingAddress billingAddress;
    @SerializedName("shipping")
    public ShippingAddress shippingAddress;

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public ShippingAddress getShippingAddress() {
        return shippingAddress;
    }

    public void setShippingAddress(ShippingAddress shippingAddress) {
        this.shippingAddress = shippingAddress;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getLastOrderId() {
        return lastOrderId;
    }

    public void setLastOrderId(String lastOrderId) {
        this.lastOrderId = lastOrderId;
    }

    public String getLastOrderDate() {
        return lastOrderDate;
    }

    public void setLastOrderDate(String lastOrderDate) {
        this.lastOrderDate = lastOrderDate;
    }

    public int getOrdersCount() {
        return ordersCount;
    }

    public void setOrdersCount(int ordersCount) {
        this.ordersCount = ordersCount;
    }

    public String getTotalSpent() {
        return totalSpent;
    }

    public void setTotalSpent(String totalSpent) {
        this.totalSpent = totalSpent;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public void setAvatarUrl(String avatarUrl) {
        this.avatarUrl = avatarUrl;
    }

    public BillingAddress getBillingAddress() {
        return billingAddress;
    }

    public void setBillingAddress(BillingAddress billingAddress) {
        this.billingAddress = billingAddress;
    }
}

File: firebasecart/src/main/java/me/gilo/woodroid/firebasecart/models/Customer.kt
package me.gilo.woodroid.firebasecart.models

import com.google.gson.annotations.SerializedName
import me.gilo.woodroid.models.BillingAddress
import me.gilo.woodroid.models.ShippingAddress

import java.io.Serializable


class Customer : Serializable {
    var id: Int = 0

    @SerializedName("created_at")
    lateinit var createdAt: String

    lateinit var email: String

    @SerializedName("first_name")
    lateinit var firstName: String

    @SerializedName("last_name")
    lateinit var lastName: String

    lateinit var username: String
    lateinit var password: String
    lateinit var role: String

    @SerializedName("last_order_id")
    lateinit var lastOrderId: String

    @SerializedName("last_order_date")
    lateinit var lastOrderDate: String

    @SerializedName("orders_count")
    var ordersCount: Int = 0

    @SerializedName("total_spent")
    lateinit var totalSpent: String

    @SerializedName("avatar_url")
    lateinit var avatarUrl: String

    @SerializedName("billing")
    lateinit var billingAddress: BillingAddress

    @SerializedName("shipping")
    lateinit var shippingAddress: ShippingAddress


}
